### PR TITLE
fix issue 148

### LINKED ISSUE #148: Auto3DSeg Learning Rate Scheduler Error when the num_interation is small (<5)
**Describe the bug**

  File "./ref_api_work_dir/segresnet2d_0/scripts/train.py", line 279, in run
    lr_scheduler.step()
  File "/opt/conda/lib/python3.8/site-packages/torch/optim/lr_scheduler.py", line 161, in step
    values = self.get_lr()
  File "/opt/conda/lib/python3.8/site-packages/torch/optim/lr_scheduler.py", line 384, in get_lr
    if (self.last_epoch == 0) or (self.last_epoch % self.step_size != 0):

**To Reproduce**
Run the Auto3DSeg tutorials on a 8-GPU system

** Error Log**

### PR DESCRIPTION
**Pull request by mingxin-zheng**
Fix #148 

Signed-off-by: Mingxin Zheng <18563433+mingxin-zheng@users.noreply.github.com>

### CODE REVIEW

File: auto3dseg/tests/test_algo_templates.py [hunk 1]
```diff
@@ -47,14 +48,22 @@
...
 sys.path.insert(0, algo_templates)
 
+num_epochs = 1
+n_data = 8
+num_gpus = 4 if torch.cuda.device_count() > 4 else torch.cuda.device_count()
+num_images_per_batch = 2
+num_epochs_per_validation = 1
+num_iterations = int(num_epochs * n_data / num_images_per_batch / num_gpus)
+len_train_loader = int(n_data / num_gpus / num_images_per_batch)
+num_iterations_per_validation = len_train_loader * num_epochs_per_validation
```
Comment (dongyang0122) [inline auto3dseg/tests/test_algo_templates.py:58]: hi @mingxin-zheng, what is the purpose for this change?

**Review by dongyang0122 — COMMENTED** [review]
Looks great!

File: auto3dseg/tests/test_algo_templates.py (hunk 1 above)
Comment (mingxin-zheng) [reply auto3dseg/tests/test_algo_templates.py:58]: Overall, this is to enhance the test.

With the previous code, some of the algorithms will fail when the `num_epochs` is set to 1, because the `num_iterations` is too small.

After the fix, the code will still work with small `num_epochs` and `num_iterations`. So that's why we have the change.

Comment (dongyang0122) [reply auto3dseg/tests/test_algo_templates.py:58]: got it. thank you!
